sdk/Sdk/FunctionMetadataGenerator.cs

```diff
@@ -378,14 +372,228 @@ private static ExpandoObject BuildBindingMetadataFromAttribute(CustomAttribute a
+                    dataType = DataType.String;
+                }
+                else if (IsOrDerivedFrom(type, Constants.IEnumerableOfBinaryType))
+                {
+                    dataType = DataType.Binary;
+                }
+                else if (isEnumerableOfT)
+                {
+                    // Find real type that "T" in IEnumerable<T> resolves to
+                    string typeName = ResolveIEnumerableOfTType(type, new Dictionary<string, string>()) ?? string.Empty;
+                    dataType = GetDataTypeFromType(typeName);
+                }
+                return true;
+            }
+
+            dataType = DataType.Undefined;
+            return false;
+        }
+
+        private static bool IsOrDerivedFrom(TypeReference type, string interfaceFullName)
+        {
+            bool isType = string.Equals(type.FullName, interfaceFullName, StringComparison.Ordinal);
+            TypeDefinition definition = type.Resolve();
+            return isType || IsDerivedFrom(definition, interfaceFullName);
+        }
+
+        private static bool IsDerivedFrom(TypeDefinition definition, string interfaceFullName)
+        {
+            var isType = string.Equals(definition.FullName, interfaceFullName, StringComparison.Ordinal);
+            return isType || HasInterface(definition, interfaceFullName) || IsSubclassOf(definition, interfaceFullName);
+        }
+
+        private static bool HasInterface(TypeDefinition definition, string interfaceFullName)
+        {
+            return definition.Interfaces.Any(i => string.Equals(i.InterfaceType.FullName, interfaceFullName, StringComparison.Ordinal));
+        }
+
+        private static bool IsSubclassOf(TypeDefinition definition, string interfaceFullName)
+        {
+            if (definition.BaseType is null)
+            {
+                return false;
+            }
+
+            TypeDefinition baseType = definition.BaseType.Resolve();
+            return IsDerivedFrom(baseType, interfaceFullName);
+        }
+
+        private static string? ResolveIEnumerableOfTType(TypeReference type, Dictionary<string, string> foundMapping)
+        {
+            // Base case: 
+            // We are at IEnumerable<T> and want to return the most recent resolution of T
+            // (Most recent is relative to IEnumerable<T>)
+            if (string.Equals(type.FullName, Constants.IEnumerableOfT, StringComparison.Ordinal))
+            {
+                if (foundMapping.TryGetValue(Constants.GenericIEnumerableArgumentName, out string typeName))
+                {
+                    return typeName;
+                }
+                else
+                {
+                    return null;
+                }
+            }
+
+            TypeDefinition definition = type.Resolve();
+            if (definition.HasGenericParameters && type is GenericInstanceType genericType)
+            {
+                for (int i = 0; i < genericType.GenericArguments.Count(); i++)
```

Comment:
For later -- Would it ever make sense to just look at the first argument, as we know `IEnumerable<T>` would always only have at most one generic arg?

Also this can be made `foreach`. As apparently, `ElementAt()` could go over the whole ienumerable every lookup depending on the implementation it says. Not that it matters much here :)

Reply:
^This was done because you might have some Foo : Bar<K, V> : IEnumerable<V> 

So in this case the recursive search would want to match V from Bar (the second generic) with V from IEnumerable (the only generic)

Reply:
Let's follow up on this case.